src/shader.cpp

```diff
@@ -0,0 +1,64 @@
+#include "shader.hpp"
+#include "globals.hpp"
+
+#include <fstream>
+#include <ios>
+#include <sys/types.h>
+
+auto BaseShader::readShaderSource_(const char *file_path) noexcept -> char * {
```

Comment:
I believe since we are using OpenGL which is a C library and it's functions are mainly C style, having to work with unique pointer will be a bit of headache, but we will see.